pkg/fission-cli/cmd/spec/list.go

```diff
@@ -60,63 +60,90 @@ func (opts *ListSubCommand) run(input cli.Input) error {
 		deployID = fr.DeploymentConfig.UID
 	}
 
-	allfn, err := getAllFunctions(opts.Client())
+	_, currentNS, err := util.GetResourceNamespace(input, flagkey.NamespaceEnvironment)
 	if err != nil {
-		return errors.Wrap(err, "error getting Functions from all namespaces")
+		return fv1.AggregateValidationErrors("Environment", err)
+	}
+
+	if input.Bool(flagkey.AllNamespaces) {
+		return opts.getResource(input, "", deployID)
+	} else {
+		return opts.getResource(input, currentNS, deployID)
+	}
+
+}
+
+func (opts *ListSubCommand) getResource(input cli.Input, namespace string, deployID string) (err error) {
+	var allfn []fv1.Function
+	printNS := namespace
+	if printNS == "" {
+		printNS = "all"
+	}
+
+	allfn, err = getAllFunctions(opts.Client(), "")
```

Comment:
We should pass the namespace parameter here?

Reply:
thank you for pointing this out.